src/main/java/io/mathdojo/Question.java

```diff
@@ -1,34 +1,56 @@
 package io.mathdojo;
 
+import java.util.Arrays;
+import java.util.Map;
+
 import org.springframework.data.annotation.Id;
 import org.springframework.data.mongodb.core.mapping.Document;
 
 
 @Document(collection = "questions")
-public class Question {
-	
+public class Question {	
+	public static Question EMPTY_DATABASE =  new Question("the-database-is-empty", null, null, null, 0, null, null, null, null, null);
+	//id must be kebab-case
 	@Id
+	private String id;
     private String questionTitle;
     private String questionBody;
     private String sampleAnswer;
-    private String[] hints;
-    private int successRate;
+    private Integer successRate;
     private String difficulty;
-    private boolean solved;
+    private String[] hints;
+    private String parentTopicTitle;
+    private String[] questionAnswerOptions;
+    private String answer;
 
     public Question() {
     }
 
-	public Question(String questionTitle, String questionBody, String sampleAnswer, String[] hints, int successRate,
-			String difficulty, boolean solved) {
+	public Question(String id, String questionTitle, String questionBody, String sampleAnswer, Integer successRate,
+			String difficulty, String[] hints, String parentTopicTitle, String[] questionAnswerOptions, String answer) {
 		super();
+		this.id = id;
 		this.questionTitle = questionTitle;
 		this.questionBody = questionBody;
 		this.sampleAnswer = sampleAnswer;
-		this.hints = hints;
 		this.successRate = successRate;
 		this.difficulty = difficulty;
-		this.solved = solved;
+		this.hints = hints;
+		this.parentTopicTitle = parentTopicTitle;
+		this.questionAnswerOptions = questionAnswerOptions;
+		this.answer = answer;
+	}
+
+	public Question(Map<String, String> queryParameters) {
+		this("the database is empty", queryParameters.get("title"), null, null, null, queryParameters.get("difficulty"), null, null, null, null);

```

Comment:
If I've understood this, you're calling the constructor when initialising a question to be updated via query parameters. That being the case, do we want to initialise it with an `id: the database is empty`